darts/models/theta.py

```diff
@@ -115,14 +120,241 @@ def predict(self, n: int) -> 'TimeSeries':
 
             replicated_seasonality = np.tile(self.seasonality.pd_series()[-self.season_period:],
                                              math.ceil(n / self.season_period))[:n]
-            if self.mode in ['multiplicative', 'mul']:
+            if self.season_mode is Season.MULTIPLICATIVE:
                 forecast *= replicated_seasonality
-            elif self.mode in ['additive', 'add']:
+            elif self.season_mode is Season.ADDITIVE:
                 forecast += replicated_seasonality
-            else:
-                raise ValueError("mode cannot be {}".format(self.mode))
 
         return self._build_forecast_series(forecast)
 
     def __str__(self):
         return 'Theta({})'.format(self.theta)
+
+
+class FourTheta(UnivariateForecastingModel):
+    def __init__(self,
+                 theta: int = 2,
+                 seasonality_period: Optional[int] = None,
+                 season_mode: Season = Season.MULTIPLICATIVE,
+                 model_mode: Model = Model.ADDITIVE,
+                 trend_mode: Trend = Trend.LINEAR,
+                 normalization: bool = True):
+        """
+        An implementation of the 4Theta method with configurable `theta` parameter.
+
+        See M4 competition `solution <https://github.com/Mcompetitions/M4-methods/blob/master/4Theta%20method.R>`_.
+
+        The training time series is de-seasonalized according to `seasonality_period`,
+        or an inferred seasonality period.
+
+        When called with `theta = 2 - X`, `model_mode = Model.ADDITIVE` and `trend_mode = Trend.LINEAR`,
+        this model is equivalent to calling `Theta(theta=X)`.
```

Comment:
Do we still need the other Theta model implementation? Wouldn't it be enough to just call this one?

Reply:
Solved after our discussion - The original Theta implementation is faster.